src/core/ext/transport/chttp2/transport/chttp2_transport.cc

```diff
@@ -716,10 +715,9 @@ static void destroy_transport_locked(void* tp, grpc_error_handle /*error*/) {
   t->memory_owner.Reset();
 }
 
-static void destroy_transport(grpc_transport* gt) {
-  grpc_chttp2_transport* t = reinterpret_cast<grpc_chttp2_transport*>(gt);
-  t->combiner->Run(GRPC_CLOSURE_CREATE(destroy_transport_locked, t, nullptr),
-                   absl::OkStatus());
+void grpc_chttp2_transport::Orphan() {
+  combiner->Run(GRPC_CLOSURE_CREATE(destroy_transport_locked, this, nullptr),
```

Comment:
Don't we need to take ref-counting into account here?  If there are still internal refs being held, we don't want to destroy the transport yet, right?  Or does this just trigger cleanup that will always cause the refs to be released?

Reply:
This just triggers the cleanup.